{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "39233e6a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://geoparse.readthedocs.io/en/latest/\n",
    "import GEOparse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "732311a1",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "04-Jan-2026 22:14:28 INFO GEOparse - Parsing D:\\Workspace_PC\\AIBI_Analytic_Tools\\data\\raw\\GSE66360_family.soft.gz: \n",
      "04-Jan-2026 22:14:28 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "04-Jan-2026 22:14:28 DEBUG GEOparse - SERIES: GSE66360\n",
      "04-Jan-2026 22:14:28 DEBUG GEOparse - PLATFORM: GPL570\n",
      "d:\\Program\\Conda\\envs\\mlwb\\Lib\\site-packages\\GEOparse\\GEOparse.py:401: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  return read_csv(StringIO(data), index_col=None, sep=\"\\t\")\n",
      "04-Jan-2026 22:14:29 DEBUG GEOparse - SAMPLE: GSM1620819\n",
      "04-Jan-2026 22:14:29 DEBUG GEOparse - SAMPLE: GSM1620820\n",
      "04-Jan-2026 22:14:29 DEBUG GEOparse - SAMPLE: GSM1620821\n",
      "04-Jan-2026 22:14:29 DEBUG GEOparse - SAMPLE: GSM1620822\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620823\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620824\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620825\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620826\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620827\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620828\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620829\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620830\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620831\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620832\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620833\n",
      "04-Jan-2026 22:14:30 DEBUG GEOparse - SAMPLE: GSM1620834\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620835\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620836\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620837\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620838\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620839\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620840\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620841\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620842\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620843\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620844\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620845\n",
      "04-Jan-2026 22:14:31 DEBUG GEOparse - SAMPLE: GSM1620846\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620847\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620848\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620849\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620850\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620851\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620852\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620853\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620854\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620855\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620856\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620857\n",
      "04-Jan-2026 22:14:32 DEBUG GEOparse - SAMPLE: GSM1620858\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620859\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620860\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620861\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620862\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620863\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620864\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620865\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620866\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620867\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620868\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620869\n",
      "04-Jan-2026 22:14:33 DEBUG GEOparse - SAMPLE: GSM1620870\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620871\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620872\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620873\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620874\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620875\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620876\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620877\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620878\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620879\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620880\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620881\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620882\n",
      "04-Jan-2026 22:14:34 DEBUG GEOparse - SAMPLE: GSM1620883\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620884\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620885\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620886\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620887\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620888\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620889\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620890\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620891\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620892\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620893\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620894\n",
      "04-Jan-2026 22:14:35 DEBUG GEOparse - SAMPLE: GSM1620895\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620896\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620897\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620898\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620899\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620900\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620901\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620902\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620903\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620904\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620905\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620906\n",
      "04-Jan-2026 22:14:36 DEBUG GEOparse - SAMPLE: GSM1620907\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620908\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620909\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620910\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620911\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620912\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620913\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620914\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620915\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620916\n",
      "04-Jan-2026 22:14:37 DEBUG GEOparse - SAMPLE: GSM1620917\n"
     ]
    }
   ],
   "source": [
    "gse = GEOparse.get_GEO(filepath=r\"D:\\Workspace_PC\\AIBI_Analytic_Tools\\data\\raw\\GSE66360_family.soft.gz\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "8c08e947",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<SERIES: GSE66360 - 99 SAMPLES, 1 d(s)>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b1fbaa08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GSM example:\n",
      "Name:  GSM1620819\n",
      "Metadata:\n",
      " - title : Muse_1_DISCOVERY_Control\n",
      " - geo_accession : GSM1620819\n",
      " - status : Public on Feb 28 2015\n",
      " - submission_date : Feb 27 2015\n",
      " - last_update_date : Feb 28 2015\n",
      " - type : RNA\n",
      " - channel_count : 1\n",
      " - source_name_ch1 : CD146+ Circulating Endothelial Cells\n",
      " - organism_ch1 : Homo sapiens\n",
      " - taxid_ch1 : 9606\n",
      " - characteristics_ch1 : disease_status: Control, cohort: DISCOVERY, cell type: CD146+ Circulating Endothelial Cells\n",
      " - treatment_protocol_ch1 : The CellTracks®AutoPrep® system was used in conjunction with the CellSearch®CEC kit and the CellSearch®profile kit (Veridex) to immunomagnetically enrich and enumerate CD146+ CECs as previously described 12. The enriched CEC samples were analyzed with the CellTracks®Analyzer II and the number of CECs in the sample determined. For CEC microarray profiling, the AutoPrep tube with the sample from the CellTracks®AutoPrep® system was removed and placed into the MagCellect Magnet for 10-min incubation. With the tube still in the MagCellect Magnet, the supernatant liquid was aspirated without disrupting the ferrofluid bound cells from which RNA was subsequently isolated.  For whole blood samples in EDTA tubes leukocytes and cellular debris was obtained for RNA isolation following RBC lysis with Erythrocyte Lysis Buffer (Qiagen, Valencia, CA).\n",
      " - growth_protocol_ch1 : Arterial blood was collected from AMI patients into both EDTA containing (Becton Dickinson, Franklin Lakes, NJ, USA) and CellSave (Veridex, Raritan, NJ, USA) tubes in the cardiac catheterization laboratory following the placement of an arterial sheath prior to the introduction of any guide wires or coronary catheters.  The samples were maintained at room temperature and processed within 36 hours of collection.\n",
      " - molecule_ch1 : total RNA\n",
      " - extract_protocol_ch1 : Enriched CEC‑derived RNA was isolated using Trizol Reagent (Life Technologies, Carlsbad, CA) according to the manufacturer's instructions\n",
      " - label_ch1 : biotin\n",
      " - label_protocol_ch1 : Purified cDNA underwent a two-step fragmentation and labeling process using the Encore Biotin Module (NuGEN).\n",
      " - hyb_protocol : The amplified cDNA targets were hybridized to Affymetrix human U133 Plus 2.0 array following protocols as suggested by the supplier (Affymetrix, Santa Clara, CA).\n",
      " - scan_protocol : Following hybridization, arrays were washed and stained using standard Affymetrix procedures before scanning on the Affymetrix GeneChip Scanner from which data was extracted using the Affymetrix Expression Console\n",
      " - data_processing : Data were processessed using the RMA normalization method, including background and quantile normalization.\n",
      " - platform_id : GPL570\n",
      " - contact_name : Eri,,Kramer\n",
      " - contact_email : ericransomkramer@gmail.com\n",
      " - contact_institute : The Scripps Research Institute\n",
      " - contact_address : 10550 N Torrey Pines Rd\n",
      " - contact_city : La Jolla\n",
      " - contact_state : CA\n",
      " - contact_zip/postal_code : 92037\n",
      " - contact_country : USA\n",
      " - supplementary_file : ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1620nnn/GSM1620819/suppl/GSM1620819_EA10113_142127_HG-U133_PLUS_2_SC1.CEL.gz\n",
      " - series_id : GSE66360\n",
      " - data_row_count : 54675\n",
      "Table data:\n",
      "      ID_REF     VALUE\n",
      "0  1007_s_at  5.866378\n",
      "1    1053_at  8.225790\n",
      "2     117_at  6.179732\n",
      "3     121_at  6.179478\n",
      "4  1255_g_at  2.792048\n",
      "\n",
      "GPL example:\n",
      "Name:  GPL570\n",
      "Metadata:\n",
      " - title : [HG-U133_Plus_2] Affymetrix Human Genome U133 Plus 2.0 Array\n",
      " - geo_accession : GPL570\n",
      " - status : Public on Nov 07 2003\n",
      " - submission_date : Nov 07 2003\n",
      " - last_update_date : Dec 14 2020\n",
      " - technology : in situ oligonucleotide\n",
      " - distribution : commercial\n",
      " - organism : Homo sapiens\n",
      " - taxid : 9606\n",
      " - manufacturer : Affymetrix\n",
      " - manufacture_protocol : see manufacturer's web site, , Complete coverage of the Human Genome U133 Set plus 6,500 additional genes for analysis of over 47,000 transcripts, All probe sets represented on the GeneChip Human Genome U133 Set are identically replicated on the GeneChip Human Genome U133 Plus 2.0 Array. The sequences from which these probe sets were derived were selected from GenBank®, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 133, April 20, 2001) and then refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the University of California, Santa Cruz Golden-Path human genome database (April 2001 release)., In addition, there are 9,921 new probe sets representing approximately 6,500 new genes. These gene sequences were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from the UniGene database (Build 159, January 25, 2003) and refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the NCBI human genome assembly (Build 31)., \n",
      " - description : Affymetrix submissions are typically submitted to GEO using the GEOarchive method described at http://www.ncbi.nlm.nih.gov/projects/geo/info/geo_affy.html, , June 03, 2009: annotation table updated with netaffx build 28, June 06, 2012: annotation table updated with netaffx build 32, June 23, 2016: annotation table updated with netaffx build 35\n",
      " - web_link : http://www.affymetrix.com/support/technical/byproduct.affx?product=hg-u133-plus, http://www.affymetrix.com/analysis/index.affx\n",
      " - contact_name : ,,Affymetrix, Inc.\n",
      " - contact_email : geo@ncbi.nlm.nih.gov, support@affymetrix.com\n",
      " - contact_phone : 888-362-2447\n",
      " - contact_institute : Affymetrix, Inc.\n",
      " - contact_address : \n",
      " - contact_city : Santa Clara\n",
      " - contact_state : CA\n",
      " - contact_zip/postal_code : 95051\n",
      " - contact_country : USA\n",
      " - contact_web_link : http://www.affymetrix.com/index.affx\n",
      " - relation : Alternative to: GPL4454 (Alternative CDF), Alternative to: GPL4866 (Alternative CDF), Alternative to: GPL5760 (Alternative CDF), Alternative to: GPL6671 (Alternative CDF), Alternative to: GPL6791 (Alternative CDF), Alternative to: GPL6879 (Alternative CDF), Alternative to: GPL7567 (Alternative CDF), Alternative to: GPL8019 (Alternative CDF), Alternative to: GPL8542 (Alternative CDF), Alternative to: GPL8715 (Alternative CDF), Alternative to: GPL8712 (Alternative CDF), Alternative to: GPL9102 (Probe Level Version), Alternative to: GPL9099 (Alternative CDF), Alternative to: GPL9101 (Alternative CDF), Alternative to: GPL9324 (Alternative CDF), Alternative to: GPL9486 (Alternative CDF), Alternative to: GPL9987 (Alternative CDF), Alternative to: GPL10175 (Alternative CDF), Alternative to: GPL10335 (Alternative CDF), Alternative to: GPL10371 (Alternative CDF), Alternative to: GPL10526 (Alternative CDF), Alternative to: GPL10881 (Alternative CDF), Alternative to: GPL10925 (Alternative CDF), Alternative to: GPL11084 (Alternative CDF), Alternative to: GPL11433 (Alternative CDF), Alternative to: GPL11670 (Alternative CDF), Alternative to: GPL13232 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 13.0.0]), Alternative to: GPL13303 (Alternative CDF.), Alternative to: GPL13916 ([Brainarray ENSG Version 14.1]), Alternative to: GPL14837 (Alternative CDF), Alternative to: GPL14877 ([Alternative CDF]), Alternative to: GPL15394 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 14.0]), Alternative to: GPL6732 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 7]), Alternative to: GPL15445 (Alternative CDF), Alternative to: GPL16066 (Alternative CDF), Alternative to: GPL16100 (Alternative CDF [GATExplorer_Ensembl v57]), Alternative to: GPL16268 (Alternative CDF [Brainarray ENSG Version 16]), Alternative to: GPL16273 (Alternative CDF [Brainarray ENSG Version 15]), Alternative to: GPL16311 (Alternative CDF [HGU133Plus2_Hs_ENTREZG Brainarray version 14.0.0]), Alternative to: GPL16356 (Alternative CDF [Brainarray HGU133Plus2_Hs_ENTREZG 15.0.0]), Alternative to: GPL16372 (Alternative CDF [Brainarray HGU133Plus2_Hs_UG v.13]), Alternative to: GPL17392 (Alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray 17.1]), Alternative to: GPL17394 (Alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray 13.0]), Alternative to: GPL17810 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 16]), Alternative to: GPL17811 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 17]), Alternative to: GPL17996 (Alternative CDF:HGU133Plus2_Hs_ENTREZG.cdf version 15.1.0), Alternative to: GPL18121 (custom CDF), Alternative to: GPL18756 (Alternative CDF [CDF: HGU133Plus2_Hs_ENTREZG_14.1.0]), Alternative to: GPL18850 (Alternative CDF), Alternative to: GPL18478 (Alternative CDF [Hs_ENTREZG_V13]), Alternative to: GPL19109 (Alternative CDF [Brainarray HGU133Plus2_Hs_ENTREZG_v18]), Alternative to: GPL19883 (Alternative CDF [Broad GenePattern]), Alternative to: GPL19918 (Gene symbol version, 10K), Alternative to: GPL20182 (Gene Symbol Version), Alternative to: GPL22321 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 20]), Alternative to: GPL22945 (alternative), Alternative to: GPL23270 ((Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 21]), Alternative to: GPL23432 (alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray version 18.0.0]), Alternative to: GPL25740 (alternative), Alternative to: GPL28718 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 23]), Alternative to: GPL29499 (Alternative CDF)\n",
      " - data_row_count : 54675\n",
      "Table data:\n",
      "          ID  GB_ACC SPOT_ID Species Scientific Name Annotation Date  \\\n",
      "0  1007_s_at  U48705     NaN            Homo sapiens     Oct 6, 2014   \n",
      "1    1053_at  M87338     NaN            Homo sapiens     Oct 6, 2014   \n",
      "2     117_at  X51757     NaN            Homo sapiens     Oct 6, 2014   \n",
      "3     121_at  X69699     NaN            Homo sapiens     Oct 6, 2014   \n",
      "4  1255_g_at  L36861     NaN            Homo sapiens     Oct 6, 2014   \n",
      "\n",
      "       Sequence Type                  Sequence Source  \\\n",
      "0  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "1  Exemplar sequence                          GenBank   \n",
      "2  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "3  Exemplar sequence                          GenBank   \n",
      "4  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "\n",
      "                                  Target Description Representative Public ID  \\\n",
      "0  U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...                   U48705   \n",
      "1  M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...                   M87338   \n",
      "2  X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...                   X51757   \n",
      "3  X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...                   X69699   \n",
      "4  L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...                   L36861   \n",
      "\n",
      "                                          Gene Title       Gene Symbol  \\\n",
      "0  discoidin domain receptor tyrosine kinase 1 //...  DDR1 /// MIR4640   \n",
      "1        replication factor C (activator 1) 2, 40kDa              RFC2   \n",
      "2               heat shock 70kDa protein 6 (HSP70B')             HSPA6   \n",
      "3                                       paired box 8              PAX8   \n",
      "4            guanylate cyclase activator 1A (retina)            GUCA1A   \n",
      "\n",
      "      ENTREZ_GENE_ID                               RefSeq Transcript ID  \\\n",
      "0  780 /// 100616237  NM_001202521 /// NM_001202522 /// NM_001202523...   \n",
      "1               5982  NM_001278791 /// NM_001278792 /// NM_001278793...   \n",
      "2               3310                                          NM_002155   \n",
      "3               7849  NM_003466 /// NM_013951 /// NM_013952 /// NM_0...   \n",
      "4               2978                         NM_000409 /// XM_006715073   \n",
      "\n",
      "                    Gene Ontology Biological Process  \\\n",
      "0  0001558 // regulation of cell growth // inferr...   \n",
      "1  0000278 // mitotic cell cycle // traceable aut...   \n",
      "2  0000902 // cell morphogenesis // inferred from...   \n",
      "3  0001655 // urogenital system development // in...   \n",
      "4  0007165 // signal transduction // non-traceabl...   \n",
      "\n",
      "                    Gene Ontology Cellular Component  \\\n",
      "0  0005576 // extracellular region // inferred fr...   \n",
      "1  0005634 // nucleus // inferred from electronic...   \n",
      "2  0005737 // cytoplasm // inferred from direct a...   \n",
      "3  0005634 // nucleus // inferred from direct ass...   \n",
      "4  0001750 // photoreceptor outer segment // infe...   \n",
      "\n",
      "                    Gene Ontology Molecular Function  \n",
      "0  0000166 // nucleotide binding // inferred from...  \n",
      "1  0000166 // nucleotide binding // inferred from...  \n",
      "2  0000166 // nucleotide binding // inferred from...  \n",
      "3  0000979 // RNA polymerase II core promoter seq...  \n",
      "4  0005509 // calcium ion binding // inferred fro...  \n"
     ]
    }
   ],
   "source": [
    "print()\n",
    "print(\"GSM example:\")\n",
    "for gsm_name, gsm in gse.gsms.items():\n",
    "    print(\"Name: \", gsm_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gsm.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print (\"Table data:\",)\n",
    "    print (gsm.table.head())\n",
    "    break\n",
    "\n",
    "print()\n",
    "print(\"GPL example:\")\n",
    "for gpl_name, gpl in gse.gpls.items():\n",
    "    print(\"Name: \", gpl_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gpl.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print(\"Table data:\",)\n",
    "    print(gpl.table.head())\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b6090f67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <td>Affymetrix Probe Set ID LINK_PRE:\"https://www....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GB_ACC</th>\n",
       "      <td>GenBank Accession Number LINK_PRE:\"http://www....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPOT_ID</th>\n",
       "      <td>identifies controls</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Species Scientific Name</th>\n",
       "      <td>The genus and species of the organism represen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Annotation Date</th>\n",
       "      <td>The date that the annotations for this probe a...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sequence Type</th>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sequence Source</th>\n",
       "      <td>The database from which the sequence used to d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Target Description</th>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Representative Public ID</th>\n",
       "      <td>The accession number of a representative seque...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Title</th>\n",
       "      <td>Title of Gene represented by the probe set.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Symbol</th>\n",
       "      <td>A gene symbol, when one is available (from Uni...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "      <td>Entrez Gene Database UID LINK_PRE:\"http://www....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RefSeq Transcript ID</th>\n",
       "      <td>References to multiple sequences in RefSeq. Th...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Ontology Biological Process</th>\n",
       "      <td>Gene Ontology Consortium Biological Process de...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Ontology Cellular Component</th>\n",
       "      <td>Gene Ontology Consortium Cellular Component de...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Ontology Molecular Function</th>\n",
       "      <td>Gene Ontology Consortium Molecular Function de...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                        description\n",
       "ID                                Affymetrix Probe Set ID LINK_PRE:\"https://www....\n",
       "GB_ACC                            GenBank Accession Number LINK_PRE:\"http://www....\n",
       "SPOT_ID                                                         identifies controls\n",
       "Species Scientific Name           The genus and species of the organism represen...\n",
       "Annotation Date                   The date that the annotations for this probe a...\n",
       "Sequence Type                                                                      \n",
       "Sequence Source                   The database from which the sequence used to d...\n",
       "Target Description                                                                 \n",
       "Representative Public ID          The accession number of a representative seque...\n",
       "Gene Title                              Title of Gene represented by the probe set.\n",
       "Gene Symbol                       A gene symbol, when one is available (from Uni...\n",
       "ENTREZ_GENE_ID                    Entrez Gene Database UID LINK_PRE:\"http://www....\n",
       "RefSeq Transcript ID              References to multiple sequences in RefSeq. Th...\n",
       "Gene Ontology Biological Process  Gene Ontology Consortium Biological Process de...\n",
       "Gene Ontology Cellular Component  Gene Ontology Consortium Cellular Component de...\n",
       "Gene Ontology Molecular Function  Gene Ontology Consortium Molecular Function de..."
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse.gpls['GPL570'].columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4f3463d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID_REF</th>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VALUE</th>\n",
       "      <td>RMA-normalized signal intensities</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              description\n",
       "ID_REF                                   \n",
       "VALUE   RMA-normalized signal intensities"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse.gsms[\"GSM1620819\"].columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7457bca8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GSM1620819',\n",
       " 'GSM1620820',\n",
       " 'GSM1620821',\n",
       " 'GSM1620822',\n",
       " 'GSM1620823',\n",
       " 'GSM1620824',\n",
       " 'GSM1620825',\n",
       " 'GSM1620826',\n",
       " 'GSM1620827',\n",
       " 'GSM1620828',\n",
       " 'GSM1620829',\n",
       " 'GSM1620830',\n",
       " 'GSM1620831',\n",
       " 'GSM1620832',\n",
       " 'GSM1620833',\n",
       " 'GSM1620834',\n",
       " 'GSM1620835',\n",
       " 'GSM1620836',\n",
       " 'GSM1620837',\n",
       " 'GSM1620838',\n",
       " 'GSM1620839',\n",
       " 'GSM1620840',\n",
       " 'GSM1620841',\n",
       " 'GSM1620842',\n",
       " 'GSM1620843',\n",
       " 'GSM1620844',\n",
       " 'GSM1620845',\n",
       " 'GSM1620846',\n",
       " 'GSM1620847',\n",
       " 'GSM1620848',\n",
       " 'GSM1620849',\n",
       " 'GSM1620850',\n",
       " 'GSM1620851',\n",
       " 'GSM1620852',\n",
       " 'GSM1620853',\n",
       " 'GSM1620854',\n",
       " 'GSM1620855',\n",
       " 'GSM1620856',\n",
       " 'GSM1620857',\n",
       " 'GSM1620858',\n",
       " 'GSM1620859',\n",
       " 'GSM1620860',\n",
       " 'GSM1620861',\n",
       " 'GSM1620862',\n",
       " 'GSM1620863',\n",
       " 'GSM1620864',\n",
       " 'GSM1620865',\n",
       " 'GSM1620866',\n",
       " 'GSM1620867',\n",
       " 'GSM1620868',\n",
       " 'GSM1620869',\n",
       " 'GSM1620870',\n",
       " 'GSM1620871',\n",
       " 'GSM1620872',\n",
       " 'GSM1620873',\n",
       " 'GSM1620874',\n",
       " 'GSM1620875',\n",
       " 'GSM1620876',\n",
       " 'GSM1620877',\n",
       " 'GSM1620878',\n",
       " 'GSM1620879',\n",
       " 'GSM1620880',\n",
       " 'GSM1620881',\n",
       " 'GSM1620882',\n",
       " 'GSM1620883',\n",
       " 'GSM1620884',\n",
       " 'GSM1620885',\n",
       " 'GSM1620886',\n",
       " 'GSM1620887',\n",
       " 'GSM1620888',\n",
       " 'GSM1620889',\n",
       " 'GSM1620890',\n",
       " 'GSM1620891',\n",
       " 'GSM1620892',\n",
       " 'GSM1620893',\n",
       " 'GSM1620894',\n",
       " 'GSM1620895',\n",
       " 'GSM1620896',\n",
       " 'GSM1620897',\n",
       " 'GSM1620898',\n",
       " 'GSM1620899',\n",
       " 'GSM1620900',\n",
       " 'GSM1620901',\n",
       " 'GSM1620902',\n",
       " 'GSM1620903',\n",
       " 'GSM1620904',\n",
       " 'GSM1620905',\n",
       " 'GSM1620906',\n",
       " 'GSM1620907',\n",
       " 'GSM1620908',\n",
       " 'GSM1620909',\n",
       " 'GSM1620910',\n",
       " 'GSM1620911',\n",
       " 'GSM1620912',\n",
       " 'GSM1620913',\n",
       " 'GSM1620914',\n",
       " 'GSM1620915',\n",
       " 'GSM1620916',\n",
       " 'GSM1620917']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gsm_name = [gsm_name for gsm_name, gsm in gse.gsms.items()]\n",
    "gsm_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "1a9f22ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>name</th>\n",
       "      <th>GSM1620819</th>\n",
       "      <th>GSM1620820</th>\n",
       "      <th>GSM1620821</th>\n",
       "      <th>GSM1620822</th>\n",
       "      <th>GSM1620823</th>\n",
       "      <th>GSM1620824</th>\n",
       "      <th>GSM1620825</th>\n",
       "      <th>GSM1620826</th>\n",
       "      <th>GSM1620827</th>\n",
       "      <th>GSM1620828</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM1620908</th>\n",
       "      <th>GSM1620909</th>\n",
       "      <th>GSM1620910</th>\n",
       "      <th>GSM1620911</th>\n",
       "      <th>GSM1620912</th>\n",
       "      <th>GSM1620913</th>\n",
       "      <th>GSM1620914</th>\n",
       "      <th>GSM1620915</th>\n",
       "      <th>GSM1620916</th>\n",
       "      <th>GSM1620917</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID_REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1007_s_at</th>\n",
       "      <td>5.866378</td>\n",
       "      <td>4.948736</td>\n",
       "      <td>5.148384</td>\n",
       "      <td>5.064419</td>\n",
       "      <td>5.017833</td>\n",
       "      <td>5.116154</td>\n",
       "      <td>5.431328</td>\n",
       "      <td>5.235271</td>\n",
       "      <td>5.773528</td>\n",
       "      <td>6.016415</td>\n",
       "      <td>...</td>\n",
       "      <td>5.419482</td>\n",
       "      <td>5.057716</td>\n",
       "      <td>5.996493</td>\n",
       "      <td>5.343133</td>\n",
       "      <td>5.558892</td>\n",
       "      <td>6.472517</td>\n",
       "      <td>5.678816</td>\n",
       "      <td>5.653286</td>\n",
       "      <td>6.013841</td>\n",
       "      <td>5.465334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1053_at</th>\n",
       "      <td>8.225790</td>\n",
       "      <td>7.525865</td>\n",
       "      <td>7.764637</td>\n",
       "      <td>7.198461</td>\n",
       "      <td>7.831450</td>\n",
       "      <td>7.203592</td>\n",
       "      <td>7.694551</td>\n",
       "      <td>7.760259</td>\n",
       "      <td>8.279814</td>\n",
       "      <td>7.580841</td>\n",
       "      <td>...</td>\n",
       "      <td>7.226348</td>\n",
       "      <td>7.105538</td>\n",
       "      <td>5.354105</td>\n",
       "      <td>8.271500</td>\n",
       "      <td>5.963346</td>\n",
       "      <td>8.261422</td>\n",
       "      <td>6.925753</td>\n",
       "      <td>7.918424</td>\n",
       "      <td>7.442701</td>\n",
       "      <td>9.225221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117_at</th>\n",
       "      <td>6.179732</td>\n",
       "      <td>6.628137</td>\n",
       "      <td>5.859151</td>\n",
       "      <td>5.974135</td>\n",
       "      <td>6.793079</td>\n",
       "      <td>6.435229</td>\n",
       "      <td>6.320546</td>\n",
       "      <td>6.483590</td>\n",
       "      <td>6.387779</td>\n",
       "      <td>6.307066</td>\n",
       "      <td>...</td>\n",
       "      <td>6.789501</td>\n",
       "      <td>7.441940</td>\n",
       "      <td>7.026929</td>\n",
       "      <td>6.003973</td>\n",
       "      <td>7.474326</td>\n",
       "      <td>6.367759</td>\n",
       "      <td>6.835756</td>\n",
       "      <td>7.577035</td>\n",
       "      <td>7.147417</td>\n",
       "      <td>6.481128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121_at</th>\n",
       "      <td>6.179478</td>\n",
       "      <td>6.582889</td>\n",
       "      <td>6.602135</td>\n",
       "      <td>6.545905</td>\n",
       "      <td>5.911542</td>\n",
       "      <td>6.285420</td>\n",
       "      <td>6.562316</td>\n",
       "      <td>6.345888</td>\n",
       "      <td>6.272806</td>\n",
       "      <td>5.971847</td>\n",
       "      <td>...</td>\n",
       "      <td>6.513437</td>\n",
       "      <td>6.317523</td>\n",
       "      <td>7.413453</td>\n",
       "      <td>6.809103</td>\n",
       "      <td>5.949091</td>\n",
       "      <td>6.440978</td>\n",
       "      <td>6.955883</td>\n",
       "      <td>6.499215</td>\n",
       "      <td>6.587065</td>\n",
       "      <td>6.897591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1255_g_at</th>\n",
       "      <td>2.792048</td>\n",
       "      <td>2.695714</td>\n",
       "      <td>2.580181</td>\n",
       "      <td>2.712124</td>\n",
       "      <td>2.723687</td>\n",
       "      <td>2.706684</td>\n",
       "      <td>2.674747</td>\n",
       "      <td>2.635150</td>\n",
       "      <td>2.696649</td>\n",
       "      <td>2.802863</td>\n",
       "      <td>...</td>\n",
       "      <td>2.924658</td>\n",
       "      <td>2.714103</td>\n",
       "      <td>3.635945</td>\n",
       "      <td>2.710944</td>\n",
       "      <td>2.979381</td>\n",
       "      <td>3.730139</td>\n",
       "      <td>2.914868</td>\n",
       "      <td>3.013901</td>\n",
       "      <td>2.729320</td>\n",
       "      <td>3.151449</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "name       GSM1620819  GSM1620820  GSM1620821  GSM1620822  GSM1620823  \\\n",
       "ID_REF                                                                  \n",
       "1007_s_at    5.866378    4.948736    5.148384    5.064419    5.017833   \n",
       "1053_at      8.225790    7.525865    7.764637    7.198461    7.831450   \n",
       "117_at       6.179732    6.628137    5.859151    5.974135    6.793079   \n",
       "121_at       6.179478    6.582889    6.602135    6.545905    5.911542   \n",
       "1255_g_at    2.792048    2.695714    2.580181    2.712124    2.723687   \n",
       "\n",
       "name       GSM1620824  GSM1620825  GSM1620826  GSM1620827  GSM1620828  ...  \\\n",
       "ID_REF                                                                 ...   \n",
       "1007_s_at    5.116154    5.431328    5.235271    5.773528    6.016415  ...   \n",
       "1053_at      7.203592    7.694551    7.760259    8.279814    7.580841  ...   \n",
       "117_at       6.435229    6.320546    6.483590    6.387779    6.307066  ...   \n",
       "121_at       6.285420    6.562316    6.345888    6.272806    5.971847  ...   \n",
       "1255_g_at    2.706684    2.674747    2.635150    2.696649    2.802863  ...   \n",
       "\n",
       "name       GSM1620908  GSM1620909  GSM1620910  GSM1620911  GSM1620912  \\\n",
       "ID_REF                                                                  \n",
       "1007_s_at    5.419482    5.057716    5.996493    5.343133    5.558892   \n",
       "1053_at      7.226348    7.105538    5.354105    8.271500    5.963346   \n",
       "117_at       6.789501    7.441940    7.026929    6.003973    7.474326   \n",
       "121_at       6.513437    6.317523    7.413453    6.809103    5.949091   \n",
       "1255_g_at    2.924658    2.714103    3.635945    2.710944    2.979381   \n",
       "\n",
       "name       GSM1620913  GSM1620914  GSM1620915  GSM1620916  GSM1620917  \n",
       "ID_REF                                                                 \n",
       "1007_s_at    6.472517    5.678816    5.653286    6.013841    5.465334  \n",
       "1053_at      8.261422    6.925753    7.918424    7.442701    9.225221  \n",
       "117_at       6.367759    6.835756    7.577035    7.147417    6.481128  \n",
       "121_at       6.440978    6.955883    6.499215    6.587065    6.897591  \n",
       "1255_g_at    3.730139    2.914868    3.013901    2.729320    3.151449  \n",
       "\n",
       "[5 rows x 99 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pivoted_control_samples = gse.pivot_samples('VALUE')[gsm_name]\n",
    "pivoted_control_samples.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebb5d544",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mlwb",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
